spec/acceptance/class_spec.rb

```diff
@@ -79,9 +79,10 @@ class { 'fail2ban':
         it { is_expected.to be_file }
       end
       describe service(service_name) do
-        if %w[stretch buster].include? fact('lsbdistcodename')
-          it { is_expected.not_to be_running }
-        else
+        it { is_expected.not_to be_running }
+        # The docker images of Debian do not use systemd, the following test
+        # cannot be performed on these images.
+        unless %w[stretch buster].include? fact('lsbdistcodename')
```

Comment:
I think @neomilium refers to this line. Please replace ```fact('lsbdistcodename')``` with ```fact('os.name')```. 

Your pull request is quite impressive, but in the future you should divide it in to several pull requests. For example:
* Add support for data in modules
* Remove dependency for lsb fact
* ...

Reply:
Switched to osfamily…